src/test/scala/edu/ie3/simona/model/participant/BMModelSpec.scala

```diff
@@ -0,0 +1,338 @@
+    UUID.fromString("bc06e089-03cd-481e-9e28-228266a148a4"),
+    "BM Model Test Type 1",
+    Quantities.getQuantity(0, edu.ie3.util.quantities.PowerSystemUnits.EURO),
+    Quantities.getQuantity(
+      0.05d,
+      edu.ie3.util.quantities.PowerSystemUnits.EURO_PER_KILOWATTHOUR,
+    ),
+    Quantities.getQuantity(
+      5,
+      edu.ie3.util.quantities.PowerSystemUnits.PERCENT_PER_HOUR,
+    ),
+    Quantities.getQuantity(
+      190,
+      edu.ie3.util.quantities.PowerSystemUnits.KILOVOLTAMPERE,
+    ),
+    1d,
+    Quantities.getQuantity(100d, tech.units.indriya.unit.Units.PERCENT),
+  )
+
+  def buildBmModel(): BMModel = {
+    new BMModel(
+      UUID.fromString("1b332f94-03e4-4abe-b142-8fceca689c53"),
+      "BM Model Test",
+      OperationInterval(0L, 86400L),
+      QControl(new CosPhiFixed("cosPhiFixed:{(0.0,1.0)}")),
+      Kilowatts(190),
+      bmType.getCosPhiRated,
+      "MockNode",
+      isCostControlled = true,
+      EUR(bmType.getOpex.getValue.doubleValue()),
+      EuroPerKilowatthour(0.51d),
+      0.05,
+    )
+  }
+
+  "A BMModel" should {
+    "calculate K1 correctly" in {
+      val bmModel = buildBmModel()
+
+      val testCases = Table(
+        ("Time", "K1 Solution"),
+        ("2019-01-04T05:15:00+01:00[Europe/Berlin]", 1d), // Friday
+        ("2019-01-07T05:15:00+01:00[Europe/Berlin]", 1d), // Monday
+        ("2019-01-05T05:15:00+01:00[Europe/Berlin]", 0.96d), // Saturday, 5:15AM
+        ("2019-01-05T15:59:00+01:00[Europe/Berlin]", 0.995d), // Sunday, 3:59PM
+      )
+
+      testCases.foreach { case (time, k1Sol) =>
+        val k1Calc = bmModel.calculateK1(ZonedDateTime.parse(time))
+        k1Calc should be(k1Sol)
+      }
+    }
+  }
+
+  "calculate K2 correctly" in {
+    val bmModel = buildBmModel()
+
+    val testCases = Table(
+      ("Time", "K2 Solution"),
+      (
+        "2019-05-29T05:15:00+02:00[Europe/Berlin]",
+        1.03d,
+      ), // Day 149 of the year
+      (
+        "2019-05-30T05:15:00+02:00[Europe/Berlin]",
+        0.61d,
+      ), // Day 150 of the year
+      (
+        "2019-08-31T05:15:00+02:00[Europe/Berlin]",
+        0.61d,
+      ), // Day 243 of the year
+      ("2019-09-01T05:15:00+02:00[Europe/Berlin]", 1.03d), // Day 244 of the year
+    )
+
+    testCases.foreach { case (time, k2Sol) =>
+      val k2Calc = bmModel.calculateK2(ZonedDateTime.parse(time))
+      k2Calc should be(k2Sol)
+    }
+  }
+
+  "calculate PTh correctly" in {
+    val bmModel = buildBmModel()
+
+    val testCases = Table(
+      ("Temperature", "K1", "K2", "PTh Sol"),
+      (19.28, 1d, 1d, 5.62d), // independent of temp
+      (30d, 2d, 3d, 33.72d),
+      (19.2799999d, 1d, 1d, 5.6147201076d), // dependent on temp
+      (15d, 1.01d, 0.61d, 6.296542d), // somewhat realistic
+    )
+
+    testCases.foreach { case (temp, k1, k2, pThSol) =>
+      val pThCalc = bmModel.calculatePTh(Celsius(temp), k1, k2)
+      pThCalc should approximate(Megawatts(pThSol))(powerTolerance)
```

Comment:
If you define only one tolerance for Power above, you can remove the implicit call of this powerTolerance here